engine/random.h

```diff
@@ -8,18 +8,22 @@ namespace pong
 class Random
 {
 public:
+    static Random& GetInstance();
+
     template<typename T>
-    static T Range(T min, T max);
-    static bool Percent(float percent);
+    static T ValueInRange(T min, T max);
+    static bool RollPercentChance(float percent);
     static bool Bool();
 
 private:
-    Random() = delete;
+    Random();
     Random(const Random&) = delete;
     Random& operator=(const Random&) = delete;
     Random(Random&&) = delete;
     Random& operator=(Random&&) = delete;
-    ~Random() = delete;
+
+    std::random_device rd;
+    std::mt19937 generator;
```

Comment:
You possibly want to init these with empty braces? just to make sure they are initialized and not garbage 

```
std::random_device rd{}
```